Editor/Scripts/DCCIntegration/Launchers/BlenderLauncher.cs

```diff
@@ -0,0 +1,244 @@
+
+        System.Diagnostics.Process m_blenderProcess;
+        static bool redirectBlenderToUnityConsole;
+
+        public RunMode runMode { get; set; }
+
+        ~BlenderLauncher()
+        {
+            Cleanup();
+        }
+
+        static string GetBlenderPath()
+        {
+            var blenderPath = EditorPrefs.GetString(editorSettingPath);
+
+            if (string.IsNullOrEmpty(blenderPath))
+            {
+                if (!DefaultAppUtility.TryGetRegisteredApplication(".blend", out blenderPath))
+                {
+                    if (Application.platform == RuntimePlatform.OSXEditor)
+                    {
+                        blenderPath = "/Applications/Blender.app/Contents/MacOS/Blender";
+                    }
+                }
+
+                SetBlenderPath(blenderPath);
+            }
+
+            return blenderPath;
+        }
+
+        public static void SetBlenderPath(string blenderPath)
+        {
+            EditorPrefs.SetString(editorSettingPath, blenderPath);
+        }
+
+        [MenuItem(menuItem_OpenInBlender, priority = 0)]
+        static void OpenBlender()
+        {
+            var selectedAsset = Selection.objects[0];
+            if (selectedAsset == null)
+                return;
+
+            var servers = UnityEngine.Object.FindObjectsOfType<MeshSyncServer>();
+
+            MeshSyncServer server = null;
+
+            foreach (var serverInScene in servers)
+            {
+                if (serverInScene.enabled)
+                {
+                    server = serverInScene;
+                    break;
+                }
+            }
+
+            if (server == null)
+            {
+                server = MeshSyncMenu.CreateMeshSyncServerMenu(null);
+            }
+
+            server.m_DCCAsset = selectedAsset;
+
+            MeshSyncServerInspectorUtils.OpenDCCAsset(server);
+        }
+
+        [MenuItem(menuItem_OpenInBlender, validate = true, priority = 0)]
+        static bool CanOpenBlender()
+        {
+            if (Selection.objects.Length == 1)
+            {
+                string assetPath = AssetDatabase.GetAssetPath(Selection.objects[0]);
+                if (assetPath.EndsWith(".blend"))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private void M_ProcessOnOutputDataReceived(object sender, System.Diagnostics.DataReceivedEventArgs e)
+        {
+            Debug.LogFormat("[BLENDER]\n{0}", e.Data);
+        }
+
+        private void M_ProcessOnErrorDataReceived(object sender, System.Diagnostics.DataReceivedEventArgs e)
+        {
+            Debug.LogErrorFormat("[BLENDER]\n{0}", e.Data);
+        }
+
+        private void M_blenderProcess_Exited(object sender, EventArgs e)
+        {
+            m_blenderProcess = null;
+        }
+
+        #region IDCCLauncher
+
+        public void OpenDCCTool(UnityEngine.Object asset)
```

Comment:
OpenDCCTool seems to work only for Blender, and therefore we need to check whether the asset is compatible before actually calling it.

Reply:
The BlenderLauncher would only have been created for a blend file, otherwise this would never get here.

Reply:
The call of this function in `MeshSyncServerInspectorUtils.OpenDCCAsset()` (which is something that still confuses me, but let's keep this in another conversation), seems to be designed that it can handle other DCCs in the future, so we should add the check here as well.

Reply:
This is the Blender launcher, it is only created for blend files. For other file types we may supporrt in the future, we would  implement another class and not get here.